docs/mkdocs/docs/aws.md

```diff
@@ -0,0 +1,73 @@
+# Getting Started with AWS S3
+
+There are detailed guides on the AWS website on S3 configuration.  You can start with the [Creating your first bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html) page.
+
+Best practices with AWS S3 can depend on your situation.  If you're looking to try ArcticDB on AWS S3 then a reasonably simple approach is the following.
+
+### 1. Create an IAM user
+
+Best practice is to manage and access S3 buckets with a Non-root IAM account in AWS.   Ideally create a new account just for trialling ArcticDB.
+
+You can create an IAM account at [IAM > Users](https://console.aws.amazon.com/iamv2/home#/users).
+- Click `Add users`.
+- Come up with a username, e.g. `arcticdbtrial`, then click `Next`.
+- Select `Attach policies directly` and attach the `AmazonS3FullAccess` policy to that account, then click `Next`.
+- Click `Create user`.
+
+### 2. Create the access key
+
+- Click on the new user in the [IAM > Users](https://console.aws.amazon.com/iamv2/home#/users) table.
+- Go to `Security credentials` > `Access keys` and click `Create access Key`
+- Select the `Local code` option, ArcticDB is the local code.
+- Check `I understand...` and click `Next`.
+- Click `Create access Key`.
+- Record the Access key and Secret access key somewhere securely.
+
+### 3. Create the bucket
+
+The rest of the steps can be performed using commands on your client machine.  Install the [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) if needed.
+
+Configure your client machine to use the new account with the Access key and Secret access key.
+You will also need to select an [AWS region](https://docs.aws.amazon.com/general/latest/gr/rande.html).
+Selecting a region that is local to your ArcticDB client will improve performance.
+```
+aws configure
+```
+
+Create a new bucket.  Bucket names must be globally unique so you will need to come up with your own.
+```
+aws s3 mb s3://arcticdbtrial
+```
+
+### 4. Connect to the bucket
+
+Using the example of `eu-west-2` for the region and `arcticdbtrial` for the bucket with authentication setup provided by `aws configure`.
+```python
+from arcticdb import Arctic
+arctic = Arctic('s3://s3.eu-west-2.amazonaws.com:arcticdbtrial?aws_auth=true')
```

Comment:
Maybe include a reasonableness check here? If basic write/read/delete/list works then we can be sure the connectivity is OK, and demote the "checking connectivity" to a troubleshooting guide?

```
>>> from arcticdb import Arctic
>>> bucket = "aseaton-20230814-arcticdbtrial"
>>> arctic = Arctic(f"s3://s3.eu-west-2.amazonaws.com:{bucket}?aws_auth=true")
>>> arctic.list_libraries()
[]
>>> arctic.create_library("test_lib")
>>> lib = arctic["test_lib"]
>>> import pandas as pd
>>> lib.write("symbol", pd.DataFrame())
VersionedItem(symbol='symbol', library='test_lib', data=n/a, version=0, metadata=None, host='S3(endpoint=s3.eu-west-2.amazonaws.com, bucket=aseaton-20230814-arcticdbtrial)')
>>> lib.read("symbol")
VersionedItem(symbol='symbol', library='test_lib', data=<class 'pandas.core.frame.DataFrame'>, version=0, metadata=None, host='S3(endpoint=s3.eu-west-2.amazonaws.com, bucket=aseaton-20230814-arcticdbtrial)')
>>> lib.read("symbol").data
Empty DataFrame
Columns: []
Index: []
>>> lib.delete("symbol")
>>> lib.list_symbols()
```

Reply:
My concern about this is that ArcticDB uses multiple backends and we can't exercise the S3 permissions individually, and so the errors may not be obvious enough.  I'll leave as an improvement for later.